src/Client/GameClient.cpp

```diff
@@ -116,6 +120,7 @@ void GameClient::update(GameMsgPlaying updateObjects) {
 void GameClient::updatePlayers(std::vector<GamePlayerPlaying> players) {
     for (GamePlayerPlaying playerUpdate: players){
         Player* player = playersMap[playerUpdate.id];
+        if (clientUsername == player->getUsername() && playerUpdate.xPos >= levelLimit) levelCompleted = true;
```

Comment:
Why is this if asking for a bool to then add it to the variable? wouldn't it be better if we put smt like:
```suggestion
        levelCompleted = clientUsername == player->getUsername() && playerUpdate.xPos >= levelLimit;
```
If your problem is that you want to set levelCompleted to true once and never change it until an external change, then maybe you would want something along the lines of:
```suggestion
        levelCompleted |= (clientUsername == player->getUsername() && playerUpdate.xPos >= levelLimit);
```
(Also this is more performant because when the level is completed, we won't ask for the second part of the equality, but is really dependant on what you imagine at the start).